src/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/General/RoAssemblyName.cs

```diff
@@ -55,7 +56,7 @@ public bool Equals(RoAssemblyName other)
                 return false;
             if (CultureName != other.CultureName)
                 return false;
-            if (!(((ReadOnlySpan<byte>)PublicKeyToken).SequenceEqual(other.PublicKeyToken)))
+            if (!PublicKeyToken.SequenceEqual(other.PublicKeyToken))
```

Comment:
I'm guessing you probably want something more efficient than linq for comparing byte arrays.